include/envoy/network/address.h

```diff
@@ -178,9 +182,9 @@ class Instance {
   virtual Type type() const PURE;
 
   /**
-   * @return name of socket interface that should be used with this address
+   * @return SocketInterface if one was configured or nullptr
```

Comment:
In what case this will be `nullptr`? Don't we have an default socket interface?

Reply:
It's `nullptr` when addresses are allocated as they have been until now, i.e., without an explicit socket interface. With the current approach, that `nullptr` is mapped to the default socket interface [here](https://github.com/envoyproxy/envoy/pull/12550/files#diff-d28ab6f8cef00963ff99f85579f7308dR61). 

We could request that all places that allocate `Address::Instance`s provide a non `nullptr` socket interface, if we think that's not too much of a hassle, i.e., all places will really have to lookup a socket interface. Another option would be to change the address instance implementations to do the work of converting `nullptr` to the default socket interface, albeit there's some dependency untangling to be done there.

Reply:
Ideally I prefer the latter, but I see there will be dependency untangling. Let's leave a TODO here then?

Reply:
Actually, I take it back. Apologies for the confusion, the second option is pretty easy to implement, so I've updated the patch accordingly. Let me know if this looks better.